scipy/spatial/qhull.pyx

```diff
@@ -1963,7 +1963,8 @@ class Delaunay(_QhullUser):
                                 if simplices[i,j] != simplices[i,k]:
                                     if setlist.add(&sets, simplices[i,j], simplices[i,k]):
                                         with gil:
-                                            raise MemoryError
+                                            setlist.free(&sets)
```

Comment:
Does the `finally:` block do this too?

Reply:
Yes it does, thanks for the review. This is no longer needed :)